tests/test_ops/test_carafe.py

```diff
@@ -1,28 +1,39 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import torch
 from torch.autograd import gradcheck
+from torch.multiprocessing import Process
+
+
+def carafe_naive_gradcheck():
+    if not torch.cuda.is_available():
+        return
+    from mmcv.ops import CARAFENaive
+    feat = torch.randn(2, 64, 3, 3, requires_grad=True, device='cuda').double()
+    mask = torch.randn(
+        2, 100, 6, 6, requires_grad=True, device='cuda').sigmoid().double()
+    gradcheck(CARAFENaive(5, 4, 2), (feat, mask), atol=1e-4, eps=1e-4)
+
+
+def carafe_gradcheck():
+    if not torch.cuda.is_available():
+        return
+    from mmcv.ops import CARAFE
+    feat = torch.randn(2, 64, 3, 3, requires_grad=True, device='cuda').double()
+    mask = torch.randn(
+        2, 100, 6, 6, requires_grad=True, device='cuda').sigmoid().double()
+    gradcheck(CARAFE(5, 4, 2), (feat, mask), atol=1e-4, eps=1e-4)
 
 
 class TestCarafe(object):
 
     def test_carafe_naive_gradcheck(self):
-        if not torch.cuda.is_available():
-            return
-        from mmcv.ops import CARAFENaive
-        feat = torch.randn(
-            2, 64, 3, 3, requires_grad=True, device='cuda').double()
-        mask = torch.randn(
-            2, 100, 6, 6, requires_grad=True,
-            device='cuda').sigmoid().double()
-        gradcheck(CARAFENaive(5, 4, 2), (feat, mask), atol=1e-4, eps=1e-4)
+        process = Process(target=carafe_naive_gradcheck)
```

Comment:
This modification seems to be weird, though I can understand the motivation.
Can we manually use `torch.cuda.empty_cache() ` to release the memory?